### ISSUE TITLE
Local mypy errors (but CI passes)

### ISSUE THREAD

**Issue by austinmatherne-wk** [Arelle/Arelle#354]
### What happened?

mypy passes in CI, but when run locally it raises 4 errors in `arelle/Cntlr.py`:

```
arelle/Cntlr.py:246: error: Unused "type: ignore" comment
arelle/Cntlr.py:548: error: Item "None" of "Optional[IO[bytes]]" has no attribute "read"  [union-attr]
arelle/Cntlr.py:552: error: Item "None" of "Optional[IO[bytes]]" has no attribute "write"  [union-attr]
arelle/Cntlr.py:553: error: Item "None" of "Optional[IO[bytes]]" has no attribute "close"  [union-attr]
```

Expected mypy to pass without any type errors on the master branch and for CI mypy results to match local runs.

Relates to #314 

### If running from the command line, what command did you run?

mypy arelle --namespace-packages

### Interface

CLI (arelleCmdLine)

### Version

f0332501cb7b774173a06e1a73e68dac61c9b632

### Download

From source (git checkout)

### Operating System

macOS 12 Monterey

**Comment by ggravlingen** [comment]
What version of Python are you running?

**Comment by austinmatherne-wk** [comment]
3.9.13

**Comment by austinmatherne-wk** [comment]
Yeah, I wiped it and also ran with a clean python environment and git checkout. It didn't help. We've also seen these errors on multiple systems with different users.

**Comment by ggravlingen** [comment]
I'm surprised I don't get those errors when I run mypy on my end. On line 548, stdout is `IO[bytes] | None`. I would have expected that we would have to add an assert here.

<img width="783" alt="image" src="https://user-images.githubusercontent.com/21142447/191832912-ffa946a6-4917-4f5d-b9f9-de852518b0ce.png">

Also, for the import of `gtk`, shouldn't we have to add that to line 566 too to be consistent?

**Comment by selgamal** [comment]
@austinmatherne-wk If I am not mistaken, the CI runs only on ubuntu, it does not run windows or mac

**Comment by austinmatherne-wk** [comment]
Thanks @selgamal. You pointed me in the right direction. I confirmed that these typing errors [occur on macOS](https://github.com/austinmatherne-wk/Arelle/actions/runs/3110527234/jobs/5041802817) (my local), but not on linux (CI).
